cmd/serve/httplib/httplib.go

```diff
@@ -201,6 +201,17 @@ func NewServer(handler http.Handler, opt *Options) *Server {
 		}
 		oldHandler := handler
 		handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			requestURI := strings.Split(r.RequestURI, "/")
+			fs.Debugf(nil, "RequestURI: %s, %d", r.RequestURI, len(requestURI))
+			if r.Method == "GET" && (len(requestURI) == 2 || (len(requestURI) > 1 && requestURI[1] == "static")) {
```

Comment:
You should check if it is webgui.

If I am running something like `./rclone serve http TestOneDrive:/ --log-level=DEBUG --htpasswd ./htpasswd`, and visit `http://127.0.0.1:8080/`, it will show my list of files without authentication - which is not what we want.

Reply:
Thanks. I'll check

Reply:
Ideally, it should redirect to the web-gui, so is a check if web-gui is being used enough?

Reply:
Weird. In my case, no auth is needed and console prints “allowed”.

Reply:
This is my dump:
```
$./rclone serve http mydrive:/ -vv --htpasswd .htpasswd 
2019/12/08 15:08:46 DEBUG : rclone: Version "v1.50.1-DEV" starting with parameters ["./rclone" "serve" "http" "mydrive:/" "-vv" "--htpasswd" ".htpasswd"]
2019/12/08 15:08:46 DEBUG : Using config file from "/home/negative0/.config/rclone/rclone.conf"
2019/12/08 15:08:46 INFO  : Using ".htpasswd" as htpasswd storage
2019/12/08 15:08:47 DEBUG : Adding path "vfs/forget" to remote control registry
2019/12/08 15:08:47 DEBUG : Adding path "vfs/refresh" to remote control registry
2019/12/08 15:08:47 DEBUG : Adding path "vfs/poll-interval" to remote control registry
2019/12/08 15:08:47 NOTICE: Google drive root '': Serving on http://127.0.0.1:8080/
2019/12/08 15:08:53 DEBUG : PICT CSI 2019-20 nomination: No export formats found for "application/vnd.google-apps.form"
2019/12/08 15:08:53 DEBUG : Enthusia '19 : Feedback form: No export formats found for "application/vnd.google-apps.form"
2019/12/08 15:08:53 DEBUG : Untitled form: No export formats found for "application/vnd.google-apps.form"
2019/12/08 15:08:53 DEBUG : Untitled form: No export formats found for "application/vnd.google-apps.form"
2019/12/08 15:08:53 DEBUG : Event Registration: No export formats found for "application/vnd.google-apps.form"
2019/12/08 15:08:53 INFO  : : 127.0.0.1:50594: Serving directory
```
Has the browser saved the password for you? Once I logged in using the password, it won't ask for the password again even after restarting rclone.

Reply:
I am getting the following,

```
» ./rclone serve http TestOneDrive:/ --log-level=DEBUG --htpasswd ./htpasswd
2019/12/08 17:45:53 DEBUG : rclone: Version "v1.50.2-DEV" starting with parameters ["./rclone" "serve" "http" "TestOneDrive:/" "--log-level=DEBUG" "--htpasswd" "./htpasswd"]
2019/12/08 17:45:53 DEBUG : Using config file from "/Users/yexiaoxing/.config/rclone/rclone.conf"
2019/12/08 17:45:53 DEBUG : One drive root '': Token expired but no uploads in progress - doing nothing
2019/12/08 17:45:53 DEBUG : TestOneDrive: Loaded invalid token from config file - ignoring
2019/12/08 17:45:53 DEBUG : TestOneDrive: Saved new token in config file
2019/12/08 17:45:54 INFO  : Using "./htpasswd" as htpasswd storage
2019/12/08 17:45:54 INFO  : One drive root '': poll-interval is not supported by this remote
2019/12/08 17:45:54 DEBUG : Adding path "vfs/forget" to remote control registry
2019/12/08 17:45:54 DEBUG : Adding path "vfs/refresh" to remote control registry
2019/12/08 17:45:54 DEBUG : Adding path "vfs/poll-interval" to remote control registry
2019/12/08 17:45:54 NOTICE: One drive root '': Serving on http://127.0.0.1:8080/
2019/12/08 17:46:00 DEBUG : RequestURI: /, 2
2019/12/08 17:46:00 DEBUG : Allowed:
2019/12/08 17:46:00 INFO  : : 127.0.0.1:61428: Serving directory
2019/12/08 17:46:07 DEBUG : RequestURI: /, 2
2019/12/08 17:46:07 DEBUG : Allowed:
2019/12/08 17:46:07 INFO  : : 127.0.0.1:61428: Serving directory
```

As you can see, there are two "Allowed" which is given by https://github.com/rclone/rclone/pull/3790/files#diff-a02d47b9e58c433f4fa481ee4eac3143R211

Reply:
I don't understand what's happening. Which OS and browser are you using? Also I am on the 1.50.1-DEV and you are on 1.50.2-DEV

Reply:
That's strange. The auth is allowing me to access directly when using "one drive" but seems to be working fine with "google-drive".

While serving the one-drive files, rclone uses a different server than the one used for google drive, and this same server is used to serve files from the web-gui.

Could we serve web-gui at something like ```localhost:5572/webgui ```?